build/Build.cs

```diff
@@ -0,0 +1,440 @@
+                  });
+
+        Target Compile =>
+            _ => _.DependsOn(CheckForbiddenWords)
+                  .DependsOn(Restore)
+                  .Executes(() =>
+                  {
+                      Log.Information("Compiling Calamari v{CalamariVersion}", NugetVersion.Value);
+                      
+                      DotNetBuild(_ => _.SetProjectFile(Solution)
+                                        .SetConfiguration(Configuration)
+                                        .SetNoRestore(true)
+                                        .SetVersion(NugetVersion.Value)
+                                        .SetInformationalVersion(GitVersionInfo?.InformationalVersion));
+                  });
+
+        Target Publish =>
+            _ => _.DependsOn(Compile)
+                  .Executes(() =>
+                  {
+                      if (!OperatingSystem.IsWindows())
+                          Log.Warning("Building Calamari on a non-windows machine will result "
+                                      + "in the {DefaultNugetPackageName} and {CloudNugetPackageName} "
+                                      + "nuget packages being built as .Net Core 3.1 packages "
+                                      + "instead of as .Net Framework 4.0 and 4.5.2 respectively. "
+                                      + "This can cause compatibility issues when running certain "
+                                      + "deployment steps in Octopus Server",
+                                      RootProjectName, $"{RootProjectName}.{FixedRuntimes.Cloud}");
+
+                      var nugetVersion = NugetVersion.Value;
+                      DoPublish(RootProjectName,
+                                OperatingSystem.IsWindows() ? Frameworks.Net40 : Frameworks.NetCoreApp31,
+                                nugetVersion);
+                      DoPublish(RootProjectName,
+                                OperatingSystem.IsWindows() ? Frameworks.Net452 : Frameworks.NetCoreApp31,
+                                nugetVersion,
+                                FixedRuntimes.Cloud);
+
+                      DoPublish(RootProjectName, Frameworks.NetCoreApp31, nugetVersion, FixedRuntimes.Portable);
+
+                      // Create the self-contained Calamari packages for each runtime ID defined in Calamari.csproj
+                      foreach (var rid in Solution?.GetProject(RootProjectName).GetRuntimeIdentifiers()!)
+                          DoPublish(RootProjectName, Frameworks.NetCoreApp31, nugetVersion, rid);
+                  });
+
+        Target PackBinaries =>
+            _ => _.DependsOn(Publish)
+                  .Executes(async () =>
+                  {
+                      var nugetVersion = NugetVersion.Value;
+                      var packageActions = new List<Action>
+                      {
+                          () => DoPackage(RootProjectName,
```

Comment:
I basically did this because Octopus Server has references to all these Calamari packages and so if you don't include them when you build the solution on non windows machines it will complain that the packages don't exist. At first I was going to update the Server .csproj to exclude them if running on non-windows but I was rightfully corrected by @shaun-od because we want to include those packages when using standard Calamari build (not a local build). So i've included them as netcore3.1 versions just to keep local builds of server happy.